linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -667,50 +667,40 @@ final class Analyzer(config: CommonPhaseConfig, initial: Boolean,
       assert(isScalaClass || isInterface,
           s"Cannot call lookupMethod($methodName) on non Scala class $this")
 
-      @tailrec
-      def tryLookupInherited(ancestorInfo: ClassInfo): Option[MethodInfo] = {
-        ancestorInfo.publicMethodInfos.get(methodName) match {
-          case Some(m) if !m.isAbstract && (!m.nonExistent || ancestorInfo == this) =>
-            Some(m)
-          case _ =>
-            ancestorInfo.superClass match {
-              case Some(superClass) => tryLookupInherited(superClass)
-              case None             => None
+      publicMethodInfos.get(methodName) match {
+        case Some(m) if !m.isAbstract => Some(m)
```

Comment:
This keeps the previous behavior w.r.t. abstract methods, but it seems this is a problem:
If we try to call a method on a class that is abstract, won't we *replace* the abstract method info with the missing method info? (similar thing with default methods, but its unclear to me ATM whether that can happen).

Reply:
That seems to be the case, yes. But it also doesn't seem to be an issue? Unless perhaps it adds *more* errors: trying to later call that method will report a missing declaration (because the abstract method is now missing). Is that what you're concerned about?

Reply:
It might add more errors, yes. My main concern here is about determinism: It would not surprise me if the exact errors here depend on the call graph traversal order. But in any case, this is not something related to this PR.